src/main/java/seedu/address/model/event/EventPeriod.java

```diff
@@ -85,6 +91,19 @@ public boolean isOverlapping(EventPeriod other) {
         return this.start.isBefore(other.end) && other.start.isBefore(this.end);
     }
 
+    /**
+     * Checks if this EventPeriod overlaps with the given start date and end date.
+     *
+     * @param start start date.
+     * @param end end date.
+     * @return True if there is an overlap, false otherwise.
+     */
+    public boolean isOverlapping(LocalDate start, LocalDate end) {
+        requireAllNonNull(start, end);
+
+        return !(this.start.toLocalDate().isAfter(end) || this.end.toLocalDate().isBefore(start));
```

Comment:
Could store both conditions as boolean variables before using them in the return statement for the sake of readability